tests/templates/test_subroutines/test_amplitude_amplification.py

```diff
@@ -0,0 +1,336 @@
+
+
+class TestDifferentiability:
+    """Test that AmplitudeAmplification is differentiable"""
+
+    @staticmethod
+    def circuit(params):
+        qml.RY(params[0], wires=0)
+        qml.AmplitudeAmplification(
+            qml.RY(params[0], wires=0),
+            qml.RZ(params[1], wires=0),
+            iters=3,
+            fixed_point=True,
+            work_wire=3,
+        )
+
+        return qml.expval(qml.PauliZ(0))
+
+    # calculated numerically with finite diff method (h = 1e-5)
+    exp_grad = np.array([-0.88109663, -0.66429297])
+
+    params = np.array([0.9, 0.1])
+
+    @pytest.mark.autograd
+    def test_qnode_autograd(self):
+        """Test that the QNode executes with Autograd."""
+
+        dev = qml.device("default.qubit")
+        qnode = qml.QNode(self.circuit, dev, interface="autograd")
+
+        params = qml.numpy.array(self.params, requires_grad=True)
+        res = qml.grad(qnode)(params)
+        print(res)
+        assert qml.math.shape(res) == (2,)
+        assert np.allclose(res, self.exp_grad, atol=1e-5)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("use_jit", [False, True])
+    @pytest.mark.parametrize("shots", [None, 50000])
+    def test_qnode_jax(self, shots, use_jit):
+        """Test that the QNode executes and is differentiable with JAX. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
+
+        dev = qml.device("default.qubit", shots=shots, seed=10)
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="jax", diff_method=diff_method)
+        if use_jit:
+            qnode = jax.jit(qnode)
+
+        params = jax.numpy.array(self.params)
+
+        jac_fn = jax.jacobian(qnode)
+        if use_jit:
+            jac_fn = jax.jit(jac_fn)
+
+        jac = jac_fn(params)
+        assert jac.shape == (2,)
+        assert np.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("shots", [None, 50000])
+    def test_qnode_torch(self, shots):
+        """Test that the QNode executes and is differentiable with Torch. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import torch
+
+        dev = qml.device("default.qubit", shots=shots, seed=10)
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="torch", diff_method=diff_method)
+
+        params = torch.tensor(self.params, requires_grad=True)
+        jac = torch.autograd.functional.jacobian(qnode, params)
+        assert qml.math.shape(jac) == (2,)
+        assert qml.math.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("shots", [None, 50000])
+    @pytest.mark.xfail(reason="tf gradient doesn't seem to be working, returns ()")
```

Comment:
@trbromley Do we need to include these tests if we are expecting them to fail and don't have any plans to fix it?

Reply:
Ups, actually maybe it works

Reply:
@Jaybsoni if I recall correctly, we'd like to keep these tests with xfail. Since the bug is (mostly?) with TensorFlow, this will provide us a signal for when we gain the capability of TF gradients for this usecase.